silkrpc/core/evm_access_list_tracer.cpp

```diff
@@ -0,0 +1,228 @@
+        << "   status: " << execution_state.status
+        << "   msg.gas: " << std::dec << execution_state.msg->gas
+        << "   msg.depth: " << std::dec << execution_state.msg->depth
+        << "}\n";
+
+    if (opcode_name == "SLOAD" && execution_state.stack.size() >= 1) {
+       const auto address = silkworm::bytes32_from_hex(intx::hex(execution_state.stack[0]));
+       add_storage(recipient, address);
+    } else if (opcode_name == "SSTORE" && execution_state.stack.size() >= 1) {
+       const auto address = silkworm::bytes32_from_hex(intx::hex(execution_state.stack[0]));
+       add_storage(recipient, address);
+    } else if ((opcode_name == "EXTCODECOPY" || opcode_name == "EXTCODEHASH" || opcode_name == "EXTCODESIZE" ||
+                opcode_name == "BALANCE" || opcode_name == "SELFDESTRUCT") &&
+                execution_state.stack.size() >= 1) {
+       const auto address = address_from_bytes32_hex(intx::hex(execution_state.stack[0]));
+       if (!exclude(address)) {
+          add_address(address);
+       }
+    } else if ((opcode_name == "DELEGATECALL" || opcode_name == "CALL" || opcode_name == "STATICCALL" || opcode_name == "CALLCODE") &&
+                execution_state.stack.size() >= 5) {
+       const auto address = address_from_bytes32_hex(intx::hex(execution_state.stack[1]));
+       if (!exclude(address)) {
+          add_address(address);
+       }
+    }
+}
+
+
+
+inline bool AccessListTracer::exclude(const evmc::address& address) {
+       // return (address == from_ || address == to_ || is_precompiled(address)); // ADD check on precompiled when available from silkworm
+       return (address == from_ || address == to_);
+}
+
+void AccessListTracer::add_storage(const evmc::address& address, const evmc::bytes32& storage) {
+    //std::cout << "add_storage:: Address: " << address << " Storage: " << storage << "\n";
+    for (int i = 0; i < access_list_.size(); i++) {
+       if (access_list_[i].account == address) {
+          for (int j = 0; j < access_list_[i].storage_keys.size(); j++) {
+            if (access_list_[i].storage_keys[j] == storage) {
+               return;
+            }
+          }
+          access_list_[i].storage_keys.push_back(storage);
+          return;
+       }
+    }
+    silkworm::AccessListEntry item;
+    item.account = address;
+    item.storage_keys.push_back(storage);
+    access_list_.push_back(item);
+}
+
+void AccessListTracer::add_address(const evmc::address& address) {
+    for (int i = 0; i < access_list_.size(); i++) {
+       if (access_list_[i].account == address) {
+          return;
+       }
+    }
+    silkworm::AccessListEntry item;
+    item.account = address;
+    access_list_.push_back(item);
+}
+
+void AccessListTracer::add_local_access_list(const std::vector<silkworm::AccessListEntry> input_access_list) {
+    for (int i = 0; i < input_access_list.size(); i++) {
+       if (!exclude(input_access_list[i].account))
```

Comment:
Add curly braces and indentation or join the two `if` conditions to improve readability

Reply:
ok